### DISCUSSION TITLE
Requesting Z-Wave NVM Backup and Restore + Radio migration in Z-Wave JS integration UI (similar to what ZHA integration has)

### DISCUSSION THREAD

**Discussion by Hedda** [zwave-js/zwave-js#7136] [Feature Requests]
### Description

This is really a frontend UI feature request for the [Z-Wave JS integration in Home Assistant ](https://www.home-assistant.io/integrations/zwave_js/)(so not for Z-Wave JS driver  specifically):

Please consider adding Z-Wave NVM Backup and Restore features as well as Z-Wave "radio migration" to the Z-Wave JS integration UI when using it with Z-Wave JS add-on instead of an external Z-Wave JS server.

I believe that these UI streamlining features will be very important for a more seamless user-experience in the GUI now that Nabu Casa is working on an upcoming official Home Assistant Z-Wave Controller dongle/stick hardware (as a new addition to the Home Assistant Connect family) that many Z-Wave integration users will probably want to migrate to, (see reference -> [https://www.home-assistant.io/blog/2024/06/12/roadmap-2024h1/#z-wave](https://www.home-assistant.io/blog/2024/06/12/roadmap-2024h1/#z-wave)).

Having those features available inside the Z-Wave JS integration UI would make the process of backing up and restoring your Z-Wave network to NVM (binary or JSON formatted) file very user-friendly. NVM Backup and MVM Restore functions are already available if using [Z-Wave JS UI](https://github.com/zwave-js/zwave-js-ui), however those NVM backup and restore feaures are not exposed in Z-Wave JS integration when just using it with the Z-Wave JS add-on.

If could first expose streamlined features in the UI for both NVM Backup and NVM Restore features to the Z-Wave integration frontend UI then the next obvious value to end-users would be to also add a UI frontend feature for migrating from an older Z-Wave 500 series radio adapter to a newer Z-Wave 700 series or Z-Wave 800 series radio adapter, and it would be awesome it there could be a built-in flow for that (just like there is for it in the ZHA integration).

Please look at frontend UI of [ZHA integration](https://www.home-assistant.io/integrations/zha) "download backup" and "migrate radio" for reference:

![image](https://github.com/user-attachments/assets/a1ae81ee-5327-475c-9133-286ca51a2c33)

If possible it would be great if the goal was to aim to have frontend feature-parity between the Z-Wave integration UI and the ZHA (Zigbee Home Automation) integration UI, at least for frontend UI user experience.

I think that all these ideas should fit well into on-going goals of making Home Assistant easier to use:

* https://www.home-assistant.io/blog/2022/01/19/streamlining-experiences/
* https://www.home-assistant.io/blog/2024/06/12/roadmap-introduction/

PS: By the way, on the backend-side ZHA uses zigpy-cli tool that save backup in JSON format, and for reference Z-Wave JS also has a "NVM editor" CLI utility that binary NVM backups from Z-Wave controllers into JSON and back:

* https://www.npmjs.com/package/@zwave-js/nvmedit

* https://github.com/zigpy/open-coordinator-backup

* https://github.com/zigpy/zigpy-cli

### Do you have a solution in mind?

Please expose more features in the frontend UI of the [Z-Wave JS integration in Home Assistant](https://www.home-assistant.io/integrations/zwave_js/) (for users not using [Z-Wave JS UI](https://github.com/zwave-js/zwave-js-ui)).

**Comment by Hedda** [comment]
Regarding Z-Wave network migration there is another alternative method when backup and restore is not an option (like example when migrating from a "legacy" controller with very old firmware to a controller with much newer firmware or vice versa), and that is for a second Z-Wave Controller to join an existing network and then take that over the Z-Wave network control from the existing Z-Wave Controller, promoting the new controller as master. That concept is discussed here:

* https://github.com/zwave-js/node-zwave-js/issues/7096

**Comment by AlCalzone** [comment]
Wouldn't this better be a feature request in the HA core repo?

**Reply by Hedda** [reply]
I do not think they accept feature requests in any repos (instead they only track bugs there) but I did post a copy of this to the Home Assistant community forums where users can discuss and vote on new ideas (though the voting is only symbolic to help indicate interest) -> https://community.home-assistant.io/t/requesting-z-wave-nvm-backup-and-restore-radio-migration-in-z-wave-js-integration-ui-similar-to-what-zha-integration-has/765599/

Posted it here too since I do not believe that any Z-Wave JS or Home Assistant frontend developers actually hang out and read any of posts there in the Home Assistant community forum(?).

**Reply by AlCalzone** [reply]
FWIW, I have this on my radar for things "needed" in HA.

**Reply by raman325** [reply]
That's right @Hedda we have a single place for feature requests to avoid sprawl so your community post was in the right place. As for who's watching the community posts, can't speak for the zwave-js team but on the HA side there aren't really zwave frontend developers, it's a lot of me with some help from others when they have bandwidth for both frontend and backend, and I am not attentive to every avenue of communication. If you are open to it, when you do create new FRs for zwave, feel free to post in the community and then DM me on Discord. I can review and add a comment telling you what its status is, either:
- won't do
- not planned (at the moment we don't plan to do it but would be willing to reconsider in the future)
- planned
- in progress
- done

Would that help?
